File: Python/Algorithm/Sorting/Insertion-Sort.py
# Inserion Sort in Python 
  
def insertionSort(arr): 
    for i in range(1, len(arr)): 
  
        key = arr[i] 
  
        # Move elementsone postion ahead as long as they're greater than key
        j = i-1
        while j >=0 and key < arr[j] : 
                arr[j+1] = arr[j] 
                j -= 1
        # Insert key
        arr[j+1] = key 


def print_array(arr):
    print ("Sorted array is:", *arr) 

  
# Driver code

n = int(input("Enter number of elements : ")) 
# Use map function to take input array  
arr = list(map(int,input("Enter the elements : ").strip().split()))[:n] 

insertionSort(arr) 
print_array(arr)

File: Java/Data Structure/insertion_sort.java
class InsertionSort {
    void sort(int arr[])
    {
        int n = arr.length;
        for (int i = 1; i < n; ++i) {
            int m = arr[i];
            int j = i - 1;

            while (j >= 0 && arr[j] > m) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = m;
        }
    }static void printArray(int arr[])
    {
        int n = arr.length;
        for (int i = 0; i < n; ++i)
            System.out.print(arr[i] + " ");

        System.out.println();
    }public static void main(String args[])
    {
        int arr[] = { 34, 1, 130, 15, 6 };
        InsertionSort ob = new InsertionSort();
        ob.sort(arr);
        printArray(arr);
    }
}
